.pre-commit-config.yaml

```
@@ -0,0 +1,40 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - name: Check TOML Syntax
+        id: check-toml
+        alias: toml
+      - name: Check YAML Syntax
+        id: check-yaml
+        alias: yaml
+      - name: Fix EOFs
+        id: end-of-file-fixer
+        alias: eof
+      - name: Strip trailing whitespace
+        id: trailing-whitespace
+        alias: whitespace
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      - name: Prevent blanket NOQAs
+        id: python-check-blanket-noqa
+        alias: noqa
+
+  - repo: local
+    hooks:
+      - name: Black
+        id: black
+        description: Runs black within the project's environment
+        entry: poetry run black . --check --diff
+        language: system
+        types: [python]
+        require_serial: true
```

The `psf/black` repo contains the pre-commit config, so we could just use that:

```suggestion
  - repo: https://github.com/psf/black
    rev: 21.10b0
    hooks:
      - id: black
        language_version: python3
```